spec/layout_test.spec.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+describe(\Dxw\Iguana\Theme\Layout::class, function () {
+	beforeEach(function () {
+		$this->layout = new \Dxw\Iguana\Theme\Layout();
+	});
+
+	afterEach(function () {
+		\Dxw\Iguana\Theme\Layout::$base = null;
+	});
+
+	it("test apply", function () {
+		$test = $this->layout::apply('x/y/z.php');
+		expect($test)->toBeAnInstanceOf(\Dxw\Iguana\Theme\Layout::class);
+		expect(\Dxw\Iguana\Theme\Layout::$wordpress_template)->toBe('x/y/z.php');
+		expect(\Dxw\Iguana\Theme\Layout::$base)->toBe('z');
+	});
+
+	it("test _toString", function () {
+		$this->layout->slug = 'slug';
+		$this->layout->templates = ['layouts/main.php'];
+
+		allow('apply_filters')->toBeCalled()
+		->with('roots_wrap_'.$this->layout->slug, $this->layout->templates)
+		->andReturn(['layouts/my-layout.php']);
+
+		allow('locate_template')
+		->toBeCalled()
+		->with(['layouts/my-layout.php'])
+		->andReturn('correct output');
+
+		expect($this->layout->__toString())->toBe('correct output');
+	});
+
+	it("test constructor", function () {
```

Comment:
The original test was:

```php
public function testConstructor()
	{
		$this->markTestIncomplete();
	}
```

although it would be good to make a better test!

The code under test here would be this:

```php
public function __construct($template = 'layouts/main.php')
	{
		$this->slug = basename($template, '.php');
		$this->templates = [$template];

		if (self::$base) {
			$str = substr($template, 0, -4);
			array_unshift($this->templates, sprintf($str.'-%s.php', self::$base));
		}
	}
```